core/src/loader.rs

```diff
@@ -52,7 +51,8 @@ pub fn load_system_program(info: &AccountInfo) -> Result<(), ProgramError> {
 /// # Returns
 /// * `Result<(), ProgramError>` - The result of the operation
 pub fn load_token_program(info: &AccountInfo) -> Result<(), ProgramError> {
```

Comment:
can you make a separate function for token22?

Reply:
added `load_token_2022_program` for checking token22 program_id,